### PR TITLE
Upgrade cryptography libraries

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes BX-####
Figma link (if any):

## What changed (plus any additional context for devs)
While this is not an ideal approach, it's the shortest path to upgrade all of our cryptography operations.
As reported via #991,  there are now better alternatives to ethers v5 (which uses elliptic) so this is the main motivation behind it.

We have considered the following approaches:
- Migrating to Ethers v6 -  It's not an easy task, doable but long and one of the main issues is its incompability with Wagmi < 1 and even after that we'd need an adapter
- MIgrating to Viem - Perf. wise seems like the best path forward.  Similar effort maybe a bit less but still a lot of work. This also requires a full upgrade of our swaps SDK since it's also using ethers v5

Both of those approaches are better in the long term (less dependencies, usage of new native browser primitives, smaller bundle size, etc) but that needs to be planned appropriately so this is a temporary security upgrade to use the best cryptography libraries available at the lowest level ( mnemonic / wallet generation and signing) ASAP

## What to test

- Onboarding via pkey
- Onboarding via seedphrase
- Send
- Swap + approvals (with permit - ex. USDC to ANY and without permit - ex. WETH to any token)
- Personal sign
- Sign Typed Data

### CODE REVIEW

**Review by BrodyHughes — APPROVED** [review]
Checked all these items:
- Onboarding via pkey
- Onboarding via seedphrase
- Send
- Swap + approvals (with permit - ex. USDC to ANY and without permit - ex. WETH to any token)
- Personal sign
- Sign Typed Data
All look good

File: src/core/keychain/RainbowSigner.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    // are made using EthereumJS instead of ethers v5
+
+    const typedTx = TransactionFactory.fromTxData({
+      data: transaction.data?.toString(),
+      to: transaction.to,
+      accessList: [],
+      maxPriorityFeePerGas: transaction.maxPriorityFeePerGas
+        ? BigNumber.from(transaction.maxPriorityFeePerGas).toHexString()
+        : undefined,
+      maxFeePerGas: transaction.maxFeePerGas
```
Comment (estebanmino) [inline src/core/keychain/RainbowSigner.ts:59]: no gasPrice for legacy supported chains?

File: src/core/keychain/keychainTypes/hdKeychain.ts [hunk 2]
```diff
@@ -87,9 +99,10 @@ export class HdKeychain implements IKeychain {
 
   getSigner(address: Address): Signer {
     const _privates = privates.get(this)!;
-    const wallet = _privates.getWalletForAddress(address);
-    if (!wallet?._isSigner) throw new Error('Not a signer');
-    return wallet;
+    const provider = getProvider({ chainId: mainnet.id });
```
Comment (estebanmino) [inline src/core/keychain/keychainTypes/hdKeychain.ts:102]: i checked l2 sends because of this, working fine, but chainId as mainnet doesn't do anything?

Comment (brunobar79) [reply src/core/keychain/keychainTypes/hdKeychain.ts:102]: The interface of signer requires to pass a provider to the constructor. That's why it's set to mainnet but when used through Wallet we always pass the right provider

File: src/core/keychain/RainbowSigner.ts (hunk 1 above)
Comment (brunobar79) [reply src/core/keychain/RainbowSigner.ts]: good call. missed that one. Just added it

File: package.json [hunk 3]
```diff
@@ -113,13 +116,15 @@
     "@vanilla-extract/sprinkles": "1.5.0",
     "@wagmi/chains": "1.6.0",
     "bignumber.js": "9.0.1",
+    "bip39": "3.1.0",
```
Comment (paulmillr) [inline package.json:119]: I believe this library is unused

**Comment by paulmillr** [comment]
looking good other than the unused dep

**Review by estebanmino — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (brunobar79) [reply package.json]: you're right! thanks for the review 🙏

**Pull request merged**